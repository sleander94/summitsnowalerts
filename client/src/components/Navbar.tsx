import { Link } from 'react-router-dom';
import { slide as Menu } from 'react-burger-menu';
import { NavbarProps } from '../types.d';
import { useState } from 'react';

const Navbar = ({ user, logout }: NavbarProps) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleIsOpen = () => {
    setOpen(!open);
  };

  const closeMenu = () => {
    setOpen(false);
  };
  return (
    <header id="header">
      <Link className="title" to="/">
        <img
          src={
            require('../assets/icons/snowy-mountains-svgrepo-com.svg').default
          }
          alt=""
        />
        <h1>Summit Snow Alerts</h1>
      </Link>
      {/* Desktop Menu */}
      <nav className="desktop">
        <Link id="home-link" className="menu-item" to="/">
          Home
        </Link>
        <Link id="mountains-link" className="menu-item" to="/conditions">
          Forecast
        </Link>
        {user && (
          <Link id="account-link" className="menu-item" to="/account">
            Account
          </Link>
        )}
        {!user && (
          <Link id="signup-link" className="menu-item" to="/signup">
            Signup
          </Link>
        )}
        {!user && (
          <Link id="login-link" className="menu-item" to="/login">
            Login
          </Link>
        )}
        {user && (
          <Link
            id="logout-link"
            className="menu-item"
            to="/"
            onClick={() => {
              logout();
            }}
          >
            Logout
          </Link>
        )}
      </nav>
      {/* Mobile Menu */}
      <Menu
        right
        width={200}
        isOpen={open}
        onOpen={handleIsOpen}
        onClose={handleIsOpen}
      >
        <Link
          id="home-link"
          className="menu-item"
          to="/"
          onClick={() => {
            closeMenu();
          }}
        >
          Home
        </Link>
        <Link
          id="mountains-link"
          className="menu-item"
          to="/conditions"
          onClick={() => {
            closeMenu();
          }}
        >
          Forecast
        </Link>
        {user && (
          <Link
            id="account-link"
            className="menu-item"
            to="/account"
            onClick={() => {
              closeMenu();
            }}
          >
            Account
          </Link>
        )}
        {!user && (
          <Link
            id="signup-link"
            className="menu-item"
            to="/signup"
            onClick={() => {
              closeMenu();
            }}
          >
            Signup
          </Link>
        )}
        {!user && (
          <Link
            id="login-link"
            className="menu-item"
            to="/login"
            onClick={() => {
              closeMenu();
            }}
          >
            Login
          </Link>
        )}
        {user && (
          <Link
            id="logout-link"
            className="menu-item"
            to="/"
            onClick={() => {
              logout();
              closeMenu();
            }}
          >
            Logout
          </Link>
        )}
      </Menu>
    </header>
  );
};

export default Navbar;
