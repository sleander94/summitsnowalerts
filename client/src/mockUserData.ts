export const mockUserData = {
  _id: 'aoifh73$%D3g3',
  name: 'Test User',
  email: 'test@email.com',
  phone: '(555)555-5555',
  password: 'aorinferu142431e5e23e',
  notifications: {
    text: true,
    email: false,
    times: {
      0: true,
      1: false,
      2: false,
      3: false,
      4: false,
      5: false,
      6: false,
      7: false,
      8: true,
      9: false,
      10: false,
      11: false,
      12: false,
      13: false,
      14: false,
      15: false,
      16: false,
      17: false,
      18: true,
      19: false,
      20: false,
      21: false,
      22: false,
      23: false,
    },
    days: {
      sunday: true,
      monday: false,
      tuesday: false,
      wednesday: false,
      thursday: true,
      friday: true,
      saturday: true,
    },
  },
  mountains: { Breckenridge: 80424, Keystone: 80435 },
};
