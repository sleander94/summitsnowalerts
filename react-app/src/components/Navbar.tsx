import { Link } from 'react-router-dom';
import { slide as Menu } from 'react-burger-menu';
import { NavbarProps } from '../types.d';
import { useState } from 'react';

const Navbar = ({ user, logout }: NavbarProps) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleIsOpen = () => {
    setOpen(!open);
  };

  const closeMenu = () => {
    setOpen(false);
  };
  return (
    <header id="header">
      <Link className="title" to="/home">
        <img
          src={
            require('../assets/icons/snowy-mountains-svgrepo-com.svg').default
          }
        />
        <h1>Summit Snow Alerts</h1>
      </Link>
      <Menu
        right
        width={200}
        isOpen={open}
        onOpen={handleIsOpen}
        onClose={handleIsOpen}
      >
        <Link
          id="home"
          className="menu-item"
          to="/home"
          onClick={() => {
            closeMenu();
          }}
        >
          Home
        </Link>
        <Link
          id="conditions"
          className="menu-item"
          to="/conditions"
          onClick={() => {
            closeMenu();
          }}
        >
          Conditions
        </Link>
        <Link
          id="account"
          className="menu-item"
          to="/account"
          onClick={() => {
            closeMenu();
          }}
        >
          Account
        </Link>
        <Link
          id="signup"
          className="menu-item"
          to="/signup"
          onClick={() => {
            closeMenu();
          }}
        >
          Signup
        </Link>
        <Link
          id="login"
          className="menu-item"
          to="/login"
          onClick={() => {
            closeMenu();
          }}
        >
          Login
        </Link>
        <button onClick={() => logout()}>Logout</button>
      </Menu>
    </header>
  );
};

export default Navbar;
