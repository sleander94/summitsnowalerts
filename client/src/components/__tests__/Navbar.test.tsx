import Navbar from '../Navbar';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mockUserData } from '../../mockUserData';
import '@testing-library/jest-dom/extend-expect';

it('Renders with signup & login buttons for mobile & desktop menus when user is undefined', () => {
  const component = render(
    <Router>
      <Navbar user={undefined} logout={() => {}} />
    </Router>
  );

  expect(screen.getAllByText('Signup')[0]).toHaveAttribute('href', '/signup');
  expect(screen.getAllByText('Signup')[1]).toHaveAttribute('href', '/signup');

  expect(screen.getAllByText('Login')[0]).toHaveAttribute('href', '/login');
  expect(screen.getAllByText('Login')[1]).toHaveAttribute('href', '/login');

  expect(component).toMatchSnapshot();
});

it('Renders with account action buttons for mobile & desktop menus when user is logged in', () => {
  const component = render(
    <Router>
      <Navbar user={mockUserData} logout={() => {}} />
    </Router>
  );

  expect(screen.getAllByText('Forecast')[0]).toHaveAttribute(
    'href',
    '/forecast'
  );
  expect(screen.getAllByText('Forecast')[1]).toHaveAttribute(
    'href',
    '/forecast'
  );

  expect(screen.getAllByText('Alerts')[0]).toHaveAttribute('href', '/alerts');
  expect(screen.getAllByText('Alerts')[1]).toHaveAttribute('href', '/alerts');

  expect(screen.getAllByText('Account')[0]).toHaveAttribute('href', '/account');
  expect(screen.getAllByText('Account')[1]).toHaveAttribute('href', '/account');

  expect(component).toMatchSnapshot();
});
