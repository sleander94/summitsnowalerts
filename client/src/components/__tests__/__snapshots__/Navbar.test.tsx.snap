// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with account action buttons for mobile & desktop menus when user is logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        id="header"
      >
        <a
          class="title"
          href="/"
        >
          <img
            alt=""
            src="snowy-mountains-svgrepo-com.svg"
          />
          <h1>
            Summit Snow Alerts
          </h1>
        </a>
        <nav
          class="desktop"
        >
          <a
            class="menu-item"
            href="/"
            id="home-link"
          >
            Home
          </a>
          <a
            class="menu-item"
            href="/forecast"
            id="forecast-link"
          >
            Forecast
          </a>
          <a
            class="menu-item"
            href="/alerts"
            id="alerts-link"
          >
            Alerts
          </a>
          <a
            class="menu-item"
            href="/account"
            id="account-link"
          >
            Account
          </a>
          <a
            class="menu-item"
            href="/"
            id="logout-link"
          >
            Logout
          </a>
        </nav>
        <div>
          <div
            class="bm-overlay"
            style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
          />
          <div>
            <div
              class="bm-burger-button"
              style="z-index: 1000;"
            >
              <button
                id="react-burger-menu-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                type="button"
              >
                Open Menu
              </button>
              <span>
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
                />
              </span>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="bm-menu-wrap"
            id=""
            style="position: fixed; right: 0px; z-index: 1100; width: 200px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
          >
            <div
              class="bm-menu"
              style="height: 100%; box-sizing: border-box; overflow: auto;"
            >
              <nav
                class="bm-item-list"
                style="height: 100%;"
              >
                <a
                  class="bm-item menu-item"
                  href="/"
                  id="home-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Home
                </a>
                <a
                  class="bm-item menu-item"
                  href="/forecast"
                  id="forecast-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Forecast
                </a>
                <a
                  class="bm-item menu-item"
                  href="/alerts"
                  id="alerts-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Alerts
                </a>
                <a
                  class="bm-item menu-item"
                  href="/account"
                  id="account-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Account
                </a>
                <a
                  class="bm-item menu-item"
                  href="/"
                  id="logout-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Logout
                </a>
              </nav>
            </div>
            <div>
              <div
                class="bm-cross-button"
                style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
              >
                <button
                  id="react-burger-cross-btn"
                  style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                  tabindex="-1"
                  type="button"
                >
                  Close Menu
                </button>
                <span
                  style="position: absolute; top: 6px; right: 14px;"
                >
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                  />
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      id="header"
    >
      <a
        class="title"
        href="/"
      >
        <img
          alt=""
          src="snowy-mountains-svgrepo-com.svg"
        />
        <h1>
          Summit Snow Alerts
        </h1>
      </a>
      <nav
        class="desktop"
      >
        <a
          class="menu-item"
          href="/"
          id="home-link"
        >
          Home
        </a>
        <a
          class="menu-item"
          href="/forecast"
          id="forecast-link"
        >
          Forecast
        </a>
        <a
          class="menu-item"
          href="/alerts"
          id="alerts-link"
        >
          Alerts
        </a>
        <a
          class="menu-item"
          href="/account"
          id="account-link"
        >
          Account
        </a>
        <a
          class="menu-item"
          href="/"
          id="logout-link"
        >
          Logout
        </a>
      </nav>
      <div>
        <div
          class="bm-overlay"
          style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
        />
        <div>
          <div
            class="bm-burger-button"
            style="z-index: 1000;"
          >
            <button
              id="react-burger-menu-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
              type="button"
            >
              Open Menu
            </button>
            <span>
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="bm-menu-wrap"
          id=""
          style="position: fixed; right: 0px; z-index: 1100; width: 200px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
        >
          <div
            class="bm-menu"
            style="height: 100%; box-sizing: border-box; overflow: auto;"
          >
            <nav
              class="bm-item-list"
              style="height: 100%;"
            >
              <a
                class="bm-item menu-item"
                href="/"
                id="home-link"
                style="display: block;"
                tabindex="-1"
              >
                Home
              </a>
              <a
                class="bm-item menu-item"
                href="/forecast"
                id="forecast-link"
                style="display: block;"
                tabindex="-1"
              >
                Forecast
              </a>
              <a
                class="bm-item menu-item"
                href="/alerts"
                id="alerts-link"
                style="display: block;"
                tabindex="-1"
              >
                Alerts
              </a>
              <a
                class="bm-item menu-item"
                href="/account"
                id="account-link"
                style="display: block;"
                tabindex="-1"
              >
                Account
              </a>
              <a
                class="bm-item menu-item"
                href="/"
                id="logout-link"
                style="display: block;"
                tabindex="-1"
              >
                Logout
              </a>
            </nav>
          </div>
          <div>
            <div
              class="bm-cross-button"
              style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
            >
              <button
                id="react-burger-cross-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                tabindex="-1"
                type="button"
              >
                Close Menu
              </button>
              <span
                style="position: absolute; top: 6px; right: 14px;"
              >
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                />
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with signup & login buttons for mobile & desktop menus when user is undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        id="header"
      >
        <a
          class="title"
          href="/"
        >
          <img
            alt=""
            src="snowy-mountains-svgrepo-com.svg"
          />
          <h1>
            Summit Snow Alerts
          </h1>
        </a>
        <nav
          class="desktop"
        >
          <a
            class="menu-item"
            href="/"
            id="home-link"
          >
            Home
          </a>
          <a
            class="menu-item"
            href="/forecast"
            id="forecast-link"
          >
            Forecast
          </a>
          <a
            class="menu-item"
            href="/signup"
            id="signup-link"
          >
            Signup
          </a>
          <a
            class="menu-item"
            href="/login"
            id="login-link"
          >
            Login
          </a>
        </nav>
        <div>
          <div
            class="bm-overlay"
            style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
          />
          <div>
            <div
              class="bm-burger-button"
              style="z-index: 1000;"
            >
              <button
                id="react-burger-menu-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                type="button"
              >
                Open Menu
              </button>
              <span>
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
                />
                <span
                  class="bm-burger-bars"
                  style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
                />
              </span>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="bm-menu-wrap"
            id=""
            style="position: fixed; right: 0px; z-index: 1100; width: 200px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
          >
            <div
              class="bm-menu"
              style="height: 100%; box-sizing: border-box; overflow: auto;"
            >
              <nav
                class="bm-item-list"
                style="height: 100%;"
              >
                <a
                  class="bm-item menu-item"
                  href="/"
                  id="home-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Home
                </a>
                <a
                  class="bm-item menu-item"
                  href="/forecast"
                  id="forecast-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Forecast
                </a>
                <a
                  class="bm-item menu-item"
                  href="/signup"
                  id="signup-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Signup
                </a>
                <a
                  class="bm-item menu-item"
                  href="/login"
                  id="login-link"
                  style="display: block;"
                  tabindex="-1"
                >
                  Login
                </a>
              </nav>
            </div>
            <div>
              <div
                class="bm-cross-button"
                style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
              >
                <button
                  id="react-burger-cross-btn"
                  style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                  tabindex="-1"
                  type="button"
                >
                  Close Menu
                </button>
                <span
                  style="position: absolute; top: 6px; right: 14px;"
                >
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                  />
                  <span
                    class="bm-cross"
                    style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      id="header"
    >
      <a
        class="title"
        href="/"
      >
        <img
          alt=""
          src="snowy-mountains-svgrepo-com.svg"
        />
        <h1>
          Summit Snow Alerts
        </h1>
      </a>
      <nav
        class="desktop"
      >
        <a
          class="menu-item"
          href="/"
          id="home-link"
        >
          Home
        </a>
        <a
          class="menu-item"
          href="/forecast"
          id="forecast-link"
        >
          Forecast
        </a>
        <a
          class="menu-item"
          href="/signup"
          id="signup-link"
        >
          Signup
        </a>
        <a
          class="menu-item"
          href="/login"
          id="login-link"
        >
          Login
        </a>
      </nav>
      <div>
        <div
          class="bm-overlay"
          style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
        />
        <div>
          <div
            class="bm-burger-button"
            style="z-index: 1000;"
          >
            <button
              id="react-burger-menu-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
              type="button"
            >
              Open Menu
            </button>
            <span>
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="bm-menu-wrap"
          id=""
          style="position: fixed; right: 0px; z-index: 1100; width: 200px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
        >
          <div
            class="bm-menu"
            style="height: 100%; box-sizing: border-box; overflow: auto;"
          >
            <nav
              class="bm-item-list"
              style="height: 100%;"
            >
              <a
                class="bm-item menu-item"
                href="/"
                id="home-link"
                style="display: block;"
                tabindex="-1"
              >
                Home
              </a>
              <a
                class="bm-item menu-item"
                href="/forecast"
                id="forecast-link"
                style="display: block;"
                tabindex="-1"
              >
                Forecast
              </a>
              <a
                class="bm-item menu-item"
                href="/signup"
                id="signup-link"
                style="display: block;"
                tabindex="-1"
              >
                Signup
              </a>
              <a
                class="bm-item menu-item"
                href="/login"
                id="login-link"
                style="display: block;"
                tabindex="-1"
              >
                Login
              </a>
            </nav>
          </div>
          <div>
            <div
              class="bm-cross-button"
              style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
            >
              <button
                id="react-burger-cross-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                tabindex="-1"
                type="button"
              >
                Close Menu
              </button>
              <span
                style="position: absolute; top: 6px; right: 14px;"
              >
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                />
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
